core/services/gateway/handlers/vault/api.go

```diff
@@ -4,19 +4,79 @@ const (
 	// Note: any addition to this list should be reflected in
 	// HandlerTypeForMethod in handler_factory.go
 	MethodSecretsCreate = "vault.secrets.create"
+	MethodSecretsGet    = "vault.secrets.get"
+	MethodSecretsUpdate = "vault.secrets.update"
 )
 
+type SecretIdentifier struct {
+	Key       string `json:"key,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
+	Owner     string `json:"owner,omitempty"`
+}
+
+func (s *SecretIdentifier) String() string {
+	return "SecretIdentifier{" +
+		"Key: " + s.Key +
+		", Namespace: " + s.Namespace +
+		", Owner: " + s.Owner +
+		"}"
+}
+
 type SecretsCreateRequest struct {
 	ID    string `json:"id"`
 	Value string `json:"value"`
+	Owner string `json:"owner"`
+}
+
+type SecretsGetRequest struct {
```

Comment:
We should align on what we use here.

In my mind we can just use the protobuf messages or copies of them (if we somehow want to diverge the internal + external representation). Did you have something else in mind?

Reply:
Yes, I think we need both for external users:
- The signed report with payload and signatures, which clients first verify
- Unwrapped raw response fields for each method, after the payload is parsed, into structured fields.

For now, I will start with the protobuf messages you have. Later when we add signature verification, we should also write utils to unwrap the raw payload into structured response fields.